pkg/kubelet/metrics/metrics.go

```diff
@@ -187,6 +188,16 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		},
 	)
+	// PLEGLastSeen is a Gauge that tracks the interval (in seconds) since the Kubelet's
+	// Pod Lifecycle Event Generator (PLEG) was last seen active.
+	PLEGLastSeen = metrics.NewGauge(
```

Comment:
It is generally preferable to report the time _at which_ an event occurred, rather than the time _since_ the event occurred.  This makes it easier to deal with latency in monitoring pipelines, since the time at which an event occurs is always correct.  The time since an event occurs is always outdated.

It also deals much better with bad kubelet health.  If the kubelet stops responding to metric scrapes, prometheus will continue to use the previously reported value for some time (usually a few minutes).  Sending the time at which an event occurred allows prometheus to alert on such cases as well.

Reply:
See https://github.com/google/cadvisor/issues/2305 for an example of where cAdvisor does this.

Reply:
Oooo interesting! Thank you - today I learned :)

Reply:
I have updated the code as you suggested.
I considered for a while whether the metric update should move to where `relistTime` is set but decided against it as my objective is to provide a metric that will give the same answer as Kubelet uses when deciding to go unhealthy.
The name remains _last_seen_ rather than, say, `last_relist`, to emphasise this.